src/cluster.cpp

```diff
@@ -138,7 +138,7 @@ getClusterReader(const Reader& zimReader, offset_t offset, CompressionType* comp
     for(blob_index_type current(m_blobReaders.size()); current<=n.v; ++current) {
       auto blobSize = getBlobSize(blob_index_t(current));
       if (blobSize.v > SIZE_MAX) {
-        m_blobReaders.push_back(std::unique_ptr<Reader>(new BufferReader(Buffer::makeBuffer(zsize_t(0)))));
+        m_blobReaders.push_back(std::unique_ptr<Reader>(new BufferReader(Buffer::makeBuffer(nullptr, zsize_t(0)))));
```

Comment:
I would add a test in `makeBuffer` and create the buffer differently if size is 0.
It seems here that is is the "constructor" (`makeBuffer`) who is broken.
I see no reason to continue creating 0 sized buffer allocating 0 sized array.

Reply:
This change is actually not part of the fix, it is just a side-improvement. For current code the `blobSize.v > SIZE_MAX` branch can be removed from `Cluster::getReader()` (or, rather replaced with an `ASSERT`).

I think you can merge the bug-fix, and then update this part as you see fit.

Reply:
I've moved the "non allocation" of 0 sized buffer in `makeBuffer`